scripts/generate_api_docs_3.js

```diff
@@ -0,0 +1,501 @@
+  `],
+  ['lib/mixins/mutable-data.js', `
+      Mixins for skipping strict dirty checking of objects and arrays.
+  `],
+
+  ['lib/mixins/element-mixin.js',  `
+      Element class mixin that provides the core API for Polymer's
+      meta-programming features. Also provides some telemetry APIs.
+  `],
+
+  ['lib/utils/array-splice.js', `
+      Computes an array of splice records indicating the minimum edits required
+      to transform the \`previous\` array into the \`current\` array.
+  `],
+  ['lib/utils/case-map.js', `
+      Module that provides utilities for converting between "dash-case"
+      and "camelCase".
+  `],
+  ['lib/utils/debounce.js', `
+      Collapse multiple callbacks into one invocation after a timer.
+  `],
+  ['lib/utils/flattened-nodes-observer.js', `
+      Class that listens for changes (additions or removals) to "flattened
+      nodes" on a given \`node\`.`
+  ],
+  ['lib/utils/flush.js', `
+      Forces several classes of asynchronously queued tasks to synchronously
+      execute.
+  `],
+  ['lib/utils/gestures.js', `
+      Module for adding cross-platform gesture event listeners.
+  `],
+  ['lib/utils/html-tag.js', `
+      A template literal tag that creates an HTML \`<template>\` element from
+      the contents of the string. This allows you to write a Polymer Template
+      in JavaScript.
+  `],
+  ['lib/utils/import-href.js', `
+      Module providing a convenience method for importing an HTML document
+      imperatively.
+  `],
+  ['lib/utils/mixin.js', `
+      Module providing utility functions for ES6 class expression mixins.
+  `],
+  ['lib/utils/path.js', `
+      Module with utilities for manipulating structured data path strings.
+  `],
+  ['lib/utils/render-status.js', `
+      Module for scheduling flushable pre-render and post-render tasks.
+  `],
+  ['lib/utils/resolve-url.js', `
+      Module with utilities for resolving URLs against a provided \`baseUri\'.
+  `],
+  ['lib/utils/settings.js', `
+      Module to read and write a small number of global configuration
+      properties.
+  `],
+  ['lib/utils/style-gather.js', `
+      Module with utilities for collection CSS text from various sources.
+  `],
+  ['lib/utils/templatize.js', `
+      Module for preparing and stamping templates that
+      utilize Polymer templating features.
+  `],
+
+]);
+
+
+/**
+ * @param {string} filename
+ * @param {FeatureIndex} featureIndex
+ */
+function getFilenameDescription(filename, featureIndex) {
+  const result = featureIndex.get(filename);
+  if (result === undefined) {
+    throw new Error(`Got no features for filename ${filename}`);
+  }
+  const features = result.allFeatures;
+  const shortName = filename.split('/').slice(2).join('/');
+  let description;
+  if (features.length === 1) {
+    const [feature] = features;
+    description = feature.summary;
+  }
+  description = description || hardcodedDescriptions.get(filename) || `TODO(write me): ${filename}`;
```

Comment:
TODO or just warn?

Maybe these debug defaults should only be written with a flag on.